frontend/iframe/viewmodels/RootViewModel.ts

```diff
@@ -157,6 +165,22 @@ export class RootViewModel extends ViewModel<SegmentType, Options> {
                 this._showLogin(loginToken);
             }
         } else {
+            if (this._singleRoomIdOrAlias) {
```

Comment:
I'm thinking maybe we could abstract this into a `getSingleRoomId()` function in this class, which I think would simplify this code here and above where we also need to resolve the room alias:

```typescript
private async getSingleRoomId(sessionId?: string): string | undefined {
    // Returns room id or undefined when not in single room mode. 
}
```

Reply:
There is also `singleRoomMode()` which returns `!!this._resolvedSingleRoomId` but it would be better if it didn't care about room alias being resolved first. Sounds like better to do in a separate PR later on.

Reply:
Separate PR makes sense 👍